services/exec/client/utils.go

```diff
@@ -45,7 +45,27 @@ func ExecRemoteCommand(ctx context.Context, conn *proxy.Conn, binary string, arg
 	return result[0].Resp, nil
 }
 
-// ExecRemoteCommand is a helper function for execing a command on one or remote hosts
+// ExecRemoteCommandAs is a helper function for execing a command on a remote host as specified user
+// using a proxy.Conn. If the conn is defined for >1 targets this will return an error.
+func ExecRemoteCommandAs(ctx context.Context, conn *proxy.Conn, user string, binary string, args ...string) (*pb.ExecResponse, error) {
```

Comment:
As discussed offline: probably better to extend existing function using opts variadic `(...) binary string, args ...string, opts ...ExecOption)`